partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1565,83 +1567,75 @@ internal class PartiQLParserDefault : PartiQLParser {
          */
 
         override fun visitOr(ctx: GeneratedParser.OrContext) = translate(ctx) {
-            convertBinaryExpr(ctx.lhs, ctx.rhs, Expr.Binary.Op.OR)
+            val l = visit(ctx.lhs) as Expr
+            val r = visit(ctx.rhs) as Expr
+            exprOr(l, r)
         }
 
         override fun visitAnd(ctx: GeneratedParser.AndContext) = translate(ctx) {
-            convertBinaryExpr(ctx.lhs, ctx.rhs, Expr.Binary.Op.AND)
+            val l = visit(ctx.lhs) as Expr
+            val r = visit(ctx.rhs) as Expr
+            exprAnd(l, r)
         }
 
         override fun visitNot(ctx: GeneratedParser.NotContext) = translate(ctx) {
             val expr = visit(ctx.exprNot()) as Expr
-            exprUnary(Expr.Unary.Op.NOT, expr)
+            exprNot(expr)
+        }
+
+        private fun checkForInvalidTokens(op: ParserRuleContext) {
```

Comment:
Added additional check for whitespace + comments between operator tokens. More context in this issue comment https://github.com/partiql/partiql-lang-kotlin/issues/1498#issuecomment-2207234803